A/argp_standalone/bundled/patches/fix-prototype-and-random.patch

```diff
@@ -0,0 +1,53 @@
+diff --git a/argp-parse.c b/argp-parse.c
+index 3a95512..ad56887 100644
+--- a/argp-parse.c
++++ b/argp-parse.c
+@@ -869,8 +869,8 @@ parser_parse_next (struct parser *parser, int *arg_ebadkey)
+    unknown option is present, EINVAL is returned; if some parser routine
+    returned a non-zero value, it is returned; otherwise 0 is returned.  */
+ error_t
+-__argp_parse (const struct argp *argp, int argc, char **argv, unsigned flags,
+-	      int *end_index, void *input)
++__argp_parse (const struct argp *__restrict argp, int argc, char **__restrict argv, unsigned flags,
++	      int *__restrict end_index, void *__restrict input)
+ {
+   error_t err;
+   struct parser parser;
+diff --git a/argp-test.c b/argp-test.c
+index d9c108d..d32d509 100644
+--- a/argp-test.c
++++ b/argp-test.c
+@@ -205,7 +205,7 @@ main (int argc, char **argv)
+ {
+   struct params params;
+   params.foonly = 0;
+-  params.foonly_default = random ();
++  params.foonly_default = 0xdeadbeef;
```

Comment:
Not very random.  Where are all these patches coming from?

Reply:
Well, random() isn't really needed at all, but it's not in the standard library so doesn't compile on windows. No randomness is needed anyways, and there's a single call to random() in the whole system, so this is just obfuscation. I could replace random() by the portable rand() / srand() but I don't see the point.